packages/headless/src/features/search/search-actions.ts

```diff
@@ -100,6 +107,27 @@ const fetchFromAPI = async (
   return {response, duration, queryExecuted, requestExecuted: request};
 };
 
+/**
+ * Prepares the search state for a search with a query by setting the query string and resetting facet and pager states.
+ * @param q (string) The new basic query expression (e.g., `acme tornado seeds`).
+ * @param enableQuerySyntax (boolean) Whether to interpret advanced [Coveo Cloud query syntax](https://docs.coveo.com/en/1814/searching-with-coveo/search-prefixes-and-operators) in the query.
+ */
+export const prepareForSearchWithQuery = createAsyncThunk<
```

Comment:
Do we want to allow customers to use this? Since it's document I guess so 🤔 

You'll have to add it in `src/features/search/search-actions-loader.ts` to do so

Reply:
Indeed, I'd prefer to go the opposite route, not exposing the action and removing the js doc here.